Finished editing the bkg sections (#25)
greenelab/knowledge-graph-review: content/03.text-mining.md

These methods involve some form of clustering or statistical calculations.
In this section we focus on methods that use unsupervised learning to extract relationships from text. 

An unsupervised extractor can exploit the fact that two entities may appear together in text.
This event is referred to as co-occurrence and studies that use this phenomenon can be found in Table {@tbl:unsupervised-methods-text-mining}.
Two databases DISEASES [@doi:10.1016/j.ymeth.2014.11.020] and STRING [@doi:10.1093/nar/gks1094] were populated using a co-occurrence scoring method on PubMed abstracts, which measured the frequency of co-mention pairs within individual sentences as well as the abstracts themselves.
This technique assumes that each individual co-occurring pair is independent from one another.
Under this assumption mention pairs that occur more than expected were presumed to implicate the presence of an association or interaction.
This approach was able to identify 543,405 disease gene associations [@doi:10.1016/j.ymeth.2014.11.020] and 792,730 high confidence protein protein interactions [@doi:10.1093/nar/gks1094], but is limited to only using PubMed abstracts.

> Full text articles are able to drastically amplify text mining power to detect relationships [@doi:10.1371/journal.pcbi.1005962; @doi:10.1093/nar/gkt1207].

Reviewer: ```suggestion
Full text articles are able to dramatically enhance relationship detection [@doi:10.1371/journal.pcbi.1005962; @doi:10.1093/nar/gkt1207].
```

Westergaard et al., used a co-occurrence approach, similar to DISEASES [@doi:10.1016/j.ymeth.2014.11.020] and STRING [@doi:10.1093/nar/gks1094], to mine full articles for protein-protein interactions and other protein related information [@doi:10.1371/journal.pcbi.1005962].
The authors discovered that full text provided better prediction power than using abstracts alone, which suggests that future text mining approaches should consider using full text to increase detection power.

> Unsupervised extractors have been focused on treating multiple biomedical relationships as multiple isolated problems.

Reviewer: ```suggestion
Unsupervised extractors often treat different biomedical relationships as multiple isolated problems.
```

> These systems repeatedly use the same model for each biomedical relationship type.

Reviewer: ```suggestion
```

In addition to single kernel methods, a recent study used an ensemble of SVMs to extract disease-gene associations [@doi:10.1371/journal.pone.0200699].
This ensemble outperformed notable disease-gene association extractors [@doi:10.1016/j.jbi.2015.08.008; @doi:10.1186/s12859-015-0472-9] in terms of precision, recall and F1 score.
Overall, SVMs have been shown to be beneficial in terms of relationship mining; however, major focus have shifted to utilizing deep learning techniques as these approaches can perform non-linear mappings of high dimensional data.

Deep learning is an increasingly popular class of techniques that can construct their own features within a high dimensional space [@raw:GoodfellowDL; @doi:10.1038/nature14539].
These methods amount to different forms of neural networks, such as recurrent or convolutional neural networks, to perform classification.

Recurrent neural networks (RNN) are designed for sequential analysis and use a repeatedly updating hidden state to make predictions.
An example of a recurrent neural network is a long short term memory (LSTM) network [@doi:10.1162/neco.1997.9.8.1735].

> Cocos et al., [@doi:10.1093/jamia/ocw180] used a LSTM to extract drug side effects from de-identified twitter posts, while Yadav et al.,[@doi:10.3233/978-1-61499-830-3-644] used an LSTM to extract protein-protein interactions.

Reviewer: ```suggestion
Cocos et al. [@doi:10.1093/jamia/ocw180] used a LSTM to extract drug side effects from de-identified twitter posts, while Yadav et al. [@doi:10.3233/978-1-61499-830-3-644] used an LSTM to extract protein-protein interactions.
```

> Other efforts have embraced LSTMs to perform relationship extraction [@arxiv:1708.03743; @doi:10.1093/bioinformatics/btw486; @arxiv:1808.09101; @doi:10.1186/s12859-017-1609-9; @doi:10.1093/jamia/ocw180]. 

Reviewer: ```suggestion
Others have also embraced LSTMs to perform relationship extraction [@arxiv:1708.03743; @doi:10.1093/bioinformatics/btw486; @arxiv:1808.09101; @doi:10.1186/s12859-017-1609-9; @doi:10.1093/jamia/ocw180]. 
```

Despite the success of these networks, training can be difficult as these networks are highly susceptible to vanishing and exploding gradients [@doi:10.1109/ICNN.1993.298725; @arxiv:1808.03314].
One proposed solution to this problem is to clip the gradients while the neural network trains [@arxiv:1211.5063].
Besides the gradient problem, these approaches only peak in performance when the datasets reach at least a tens of thousand of data points [@arxiv:1707.02968].

Convolutional neural networks (CNNs), which are widely applied for image analysis, use multiple kernel filters to capture small subsets of an overall image [@doi:10.1038/nature14539].
In the context of text mining an image is replaced with words within a sentence mapped to dense vectors (i.e., word embeddings) [@arxiv:1301.3781; @arxiv:1310.4546].

> Peng et al.,  used a CNN to extract sentences that mentioned protein-protein interactions [@arxiv:1706.01556v2] and Zhou et al., used a CNN to extract chemical-disease relations [@doi:10.1186/s12859-019-2873-7].

Reviewer: ```suggestion
Peng et al.  used a CNN to extract sentences that mentioned protein-protein interactions [@arxiv:1706.01556v2] and Zhou et al. used a CNN to extract chemical-disease relations [@doi:10.1186/s12859-019-2873-7].
```

> Other works have utilized CNNs and variants of CNNs to extract relationships from text [@doi:10.1177/0165551516673485; @doi:10.1093/database/bay073; @doi:10.1101/730085].

Reviewer: ```suggestion
Others have used CNNs and variants of CNNs to extract relationships from text [@doi:10.1177/0165551516673485; @doi:10.1093/database/bay073; @doi:10.1101/730085].
```

Weak or distant supervision takes a different approach that uses noisy or even erroneous labels to train classifiers [@doi:10.3115/1690219.1690287; @doi:10.1093/bioinformatics/btv476; @doi:10.14778/3157794.3157797; @doi:10.1145/3209889.3209898].
Under this paradigm sentences are labeled based on their mention pair being present (positive) or absent (negative) in a database and, once labeled, machine learning classifier can now be trained to extract relationships from text [@doi:10.3115/1690219.1690287].
For example, Thomas et al., [@raw:Thomas2011LearningTE] used distant supervision to train a SVM to extract sentences mentioning protein-protein interactions (ppi). 
Their SVM model achieved comparable performance against a baseline model; however, the noise generated via distant supervision was difficult to eradicate [@raw:Thomas2011LearningTE].
A number of efforts have focused on combining distant supervision with other types of labeling strategies to mitigate the negative impacts of noisy knowledge bases [@arxiv:1805.09927; @arxiv:1805.09929; @doi:10.18653/v1/W17-2323].

> Nicholson et al. [@doi:10.1101/730085] found that, in some circumstances, these strategies can be reused across different types of biomedical relationships to learn a heterogeneous knowledge graph given that those relationships describe similar physical concepts.

Reviewer: ```suggestion
Nicholson et al. [@doi:10.1101/730085] found that, in some circumstances, these strategies can be reused across different types of biomedical relationships to learn a heterogeneous knowledge graph in cases where those relationships describe similar physical concepts.
```
